File: IQ/Classes/clsUserState.vb
<Serializable>
Public Class clsUserState
    Public lid As String
    Public root As String
    Public path As String
    Public QuoteID As Integer?
    Public foci As String 'is a CD list
    Public treeCursorPath As String
    Public branchStates As List(Of KeyValuePair(Of String, clsBranchState))
    Public AgentAccount As Integer
    Public BuyerAccount As Integer
    Public ScreenHeaders As List(Of clsScreenHeaderState)
    Public showOnly As Integer  'Render only this (system) branch (formerly 'configuring')
    Public Paradigm As enumParadigm
    Public mopUpvalues As List(Of KeyValuePair(Of Object, Object))
End Class

Public Class clsScreenHeaderState
    Public Path As String
    Public QuickFiltersVisible As Boolean
    Public Filters As List(Of KeyValuePair(Of Integer, List(Of KeyValuePair(Of clsFilter, List(Of Int64))))) 'As List(Of KeyValuePair(Of clsField, List(Of KeyValuePair(Of clsFilter, List(Of Int64)))))
    Public Sorts As List(Of KeyValuePair(Of Integer, clsPriorityDirection))
End Class

File: IQ/Classes/clsUserState.cs
[Serializable]
public class clsUserState
{
    public string lid;
    public string root;
    public string path;
    public int? QuoteID;
    public string foci; //is a CD list
    public string treeCursorPath;
    public List<KeyValuePair<string, clsBranchState>> branchStates;
    public int AgentAccount;
    public int BuyerAccount;
    public List<clsScreenHeaderState> ScreenHeaders;
    public int showOnly; //Render only this (system) branch (formerly 'configuring')
    public enumParadigm Paradigm;
    public List<KeyValuePair<object, object>> mopUpvalues;
}

public class clsScreenHeaderState
{
    public string Path;
    public bool QuickFiltersVisible;
    public List<KeyValuePair<int, List<KeyValuePair<clsFilter, List<long>>>>> Filters; //As List(Of KeyValuePair(Of clsField, List(Of KeyValuePair(Of clsFilter, List(Of Int64)))))
    public List<KeyValuePair<int, clsPriorityDirection>> Sorts;
}